<robot name="compi">
    <link name="base_link"/>
    <link name="link1"/>
    <link name="link2"/>
    <link name="link3"/>
    <link name="link4"/>
    <link name="link5"/>
    <link name="link6"/>
    <link name="tcp"/>

    <joint name="joint1" type="revolute">
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <limit effort="99" velocity="99" lower="-2.094" upper="2.094"/>
      <parent link="base_link"/>
      <child link="link1"/>
      <axis xyz="0 0 1.0"/>
    </joint>

    <joint name="joint2" type="revolute">
      <origin xyz="0 0 0.158" rpy="1.570796 0 0"/>
      <limit effort="99" velocity="99" lower="0.17079632679489665" upper="2.9707963267948965"/>
      <parent link="link1"/>
      <child link="link2"/>
      <axis xyz="0 0 1.0"/>
    </joint>

    <joint name="joint3" type="revolute">
      <origin xyz="0.28 0 0" rpy="0 0 0"/>
      <limit effort="99" velocity="99" lower="-3.140796326794897" upper="-0.0007963267948964958"/>
      <parent link="link2"/>
      <child link="link3"/>
      <axis xyz="0 0 1.0"/>
    </joint>

    <joint name="joint4" type="revolute">
      <origin xyz="0 0.34 0" rpy="-1.570796 0 0"/>
      <limit effort="99" velocity="99" lower="-2.1" upper="2.1"/>
      <parent link="link3"/>
      <child link="link4"/>
      <axis xyz="0 0 1.0"/>
    </joint>

    <joint name="joint5" type="revolute">
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <limit effort="99" velocity="99" lower="-2.1" upper="2.1"/>
      <parent link="link4"/>
      <child link="link5"/>
      <axis xyz="0 0 1.0"/>
    </joint>

    <joint name="joint6" type="revolute">
      <origin xyz="0 0 0" rpy="-1.570796 0 0"/>
      <limit effort="99" velocity="99" lower="-2.1" upper="2.1"/>
      <parent link="link5"/>
      <child link="link6"/>
      <axis xyz="0 0 1.0"/>
    </joint>

    <joint name="jointtcp" type="fixed">
      <origin xyz="0 0 0.346" rpy="0 0 0"/>
      <parent link="link6"/>
      <child link="tcp"/>
    </joint>
</robot>